src/0x.ts

```diff
@@ -163,6 +163,12 @@ export class ZeroEx {
      * @return  An instance of the 0x.js ZeroEx class.
      */
     constructor(provider: Web3Provider, config?: ZeroExConfig) {
+        if (_.isUndefined((provider as any).sendAsync)) {
```

Comment:
If the user passes in something that is **not** a provider, it also won't have the `sendAsync` method. It will also not have the `send` method. We need to also check if it has the `send` method, otherwise we want to throw a helpful error message about how an invalid provider was supplied. Maybe:

```
Invalid `provider` supplied to ZeroEx constructor. A provider must have a `send` or `sendAsync` method in order to be valid.
```